xbmc/Application.cpp

```diff
@@ -3173,7 +3173,15 @@ void CApplication::OnPlayBackStarted(const CFileItem &file)
   CLog::LogF(LOGDEBUG,"CApplication::OnPlayBackStarted");
 
   // check if VideoPlayer should set file item stream details from its current streams
-  if (file.GetProperty("get_stream_details_from_player").asBoolean())
+  if (file.GetProperty("get_stream_details_from_player").asBoolean()
+      || file.IsDiscImage()
+      || file.IsDVDFile()
+      || file.IsBDFile()
+      || file.IsBluray()
+      || file.IsDVD()
+      || file.IsOnDVD()
+      || !file.HasVideoInfoTag()
+      || !file.GetVideoInfoTag()->HasStreamDetails())
```

Comment:
After some more thought I *think* the 2 checks above need to go. This conflicts with the thumb/details extractor in the thumbloader. Now the details could be extracted here before the thumbloader does, which, as I recall, could give problems with eg. wrong AR (see #15934) . Not sure though if this is still the case.

Reply:
OK.  That still won't prevent overwriting details from nfo files associated with iso's or dvd/bluray directories though, as mentioned in that issue.

Reply:
Ok, than I think the safe approach would be (the other way around):
if (file.GetProperty("get_stream_details_from_player").asBoolean()
      || ((!file.HasVideoInfoTag() || !file.GetVideoInfoTag()->HasStreamDetails())
      && (file.IsDiscImage()
      ||  file.IsDVDFile()
      ||  file.IsBDFile()
      ||  file.IsBluray()
      ||  file.IsDVD()
      ||  file.IsOnDVD()))

Reply:
@arnova LGTM.  If a user has gone to the trouble of adding the streamdetails to an nfo, we should not over-write that.  The only (small) issue I can foresee now is if a user happens to play the wrong audio track (eg 2 channel vs 5.1).  That will now stay set but can be worked around by removing and re-adding the iso and in any case it will probably only be a tiny minority that need to do that.